ext/standard/string.c

```diff
@@ -5737,6 +5737,29 @@ PHP_FUNCTION(substr_count)
 }
 /* }}} */
 
+static zend_always_inline void php_str_pad_fill(zend_string *result, size_t pad_chars, const char *pad_str, size_t pad_str_len) {
+	char *dest = ZSTR_VAL(result) + ZSTR_LEN(result);
+	
+	if (pad_str_len == 1) {
+		memset(dest, pad_str[0], pad_chars);
+
+		return;
+	}
+
+	size_t full_copies = pad_chars / pad_str_len;
+	size_t remainder = pad_chars % pad_str_len;
+	char *p = dest;
+
+	for (size_t j = 0; j < full_copies; ++j) {
+		memcpy(p, pad_str, pad_str_len);
+		p += pad_str_len;
+	}
+
+	if (remainder > 0) {
+		memcpy(p, pad_str, remainder);
+	}
```

Comment:
We can avoid the modulo entirely (untested):
```suggestion
	char *p = dest;
	char *end = p + pad_chars;

	while (p < dest - pad_str_len) {
		p = zend_mempcpy(p, pad_str, pad_str_len);
	}
	
	size_t remainder = dest - p;
	memcpy(p, pad_str, remainder);
```

Reply:
I adapted your snippet and pushed a new version. Indeed the modulo can be avoided, thanks!